aws/session/session.go

```diff
@@ -12,87 +12,56 @@ import (
 
 // NewPoolSafe is used to create a pool of AWS sessions with different profile/region permutations
 func NewPoolSafe(profiles []string, regions []string, logger *log.Logger) (allSessions *PoolSafe) {
+	sessions := map[string]*Pool{}
```

Comment:
Is it correct to assume that you're removing the concurrent creation because we're no longer validating the sessions at the time of creation?

Since this was (I believe) the only reason for the threadsafe PoolSafe type, we should consider removing it and update things to use a `map[string]Pool` (or whatever the name of Pool ends up being).

Reply:
Yeah without it there really isn't value considering how quickly it happens. Still value in using our own type. Will provide a mockable interface when we get to that point for better tests of the core part of the commands.